src/whitenoise/media_manager/mod.rs

```diff
@@ -0,0 +1,356 @@
+//! Media handling module for the Whitenoise application.
+//!
+//! This module provides functionality for handling media files in the application:
+//! - File encryption and decryption using ChaCha20-Poly1305
+//! - File upload to the Blossom server
+//! - Local caching of media files
+//! - Generation of IMETA tags for Nostr events
+//! - Image processing and metadata extraction
+//! - Media sanitization and security checks
+//!
+//! The module is designed to work with the following workflow:
+//! 1. Files are sanitized to remove sensitive metadata
+//! 2. Files are encrypted before upload
+//! 3. Encrypted files are uploaded to Blossom
+//! 4. Original files are cached locally
+//! 5. IMETA tags are generated for Nostr events
+//!
+//! # Security
+//!
+//! All files are encrypted using ChaCha20-Poly1305 before upload to ensure
+//! end-to-end encryption. The encryption key is derived from the exporter secret.
+//! Files are also sanitized to remove potentially sensitive metadata before being
+//! processed or stored.
+//!
+//! # Caching
+//!
+//! Files are cached locally to improve performance and reduce bandwidth usage.
+//! The cache is organized by MLS group ID and uses SHA256 hashes for file identification.
+//!
+//! # IMETA Tags
+//!
+//! IMETA tags are generated for Nostr events containing:
+//! - File URL
+//! - MIME type
+//! - Original filename
+//! - For images: dimensions and blurhash
+//! - SHA256 hash of the original file
+//! - Decryption information (nonce and algorithm)
+
+mod cache;
+mod encryption;
+mod errors;
+mod sanitizer;
+mod types;
+
+use anyhow::anyhow;
+pub use errors::MediaError;
+use nostr_blossom::client::BlossomClient;
+use nwc::nostr::hashes::Hash;
+pub use sanitizer::sanitize_media;
+pub use types::*;
+
+use nostr_mls::prelude::*;
+use nostr_sdk::hashes::sha256::Hash as Sha256Hash;
+use sha2::{Digest, Sha256};
+
+use crate::whitenoise::accounts::Account;
+use crate::whitenoise::error::{BlossomError, Result};
+use crate::whitenoise::media_manager::encryption::decrypt_file;
+use crate::{MessageWithTokens, Whitenoise, WhitenoiseError};
+
+pub struct MediaManager {
+    blossom: BlossomClient,
+}
+
+impl Default for MediaManager {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl MediaManager {
+    pub fn new() -> Self {
+        let blossom = if cfg!(debug_assertions) {
+            BlossomClient::new(Url::parse("http://localhost:3000").unwrap())
+        } else {
+            BlossomClient::new(Url::parse("https://blossom.primal.net/").unwrap())
+        };
+        Self { blossom }
+    }
```

Comment:
I think we should have the `new` method be able to receive the BlossomClient "from outside" (dependency injection) and have the default call new with the actual default.

This will allow us to more easily test things, and also I think will make it easier to use the MediaManager with multiple/other blossom servers in the future. What do you think?